internal/graph/check.go

```diff
@@ -21,15 +21,16 @@ var tracer = otel.Tracer("internal/graph/check")
 // CheckResolver represents an interface that can be implemented to provide recursive resolution
 // of a Check.
 type CheckResolver interface {
-	ResolveCheck(ctx context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error)
+	execute(ctx context.Context, req *ResolveCheckRequest) (*ResolveCheckResponse, error)
```

Comment:
Please do not change this. It is purposefully designed as is to match a gRPC style stub so that it can be implemented as an RPC dispatcher.